impeller/renderer/backend/vulkan/context_vk.h

```diff
@@ -136,6 +136,14 @@ class ContextVK final : public Context,
 
   std::shared_ptr<FenceWaiterVK> GetFenceWaiter() const;
 
+  void MarkRasterThread() const {
+    raster_thread_id_ = std::this_thread::get_id();
+  }
+
+  bool IsOnRasterThread() const {
```

Comment:
I don't think it is a given that we will create a context on the raster thread. If we need a dedicated thread for the allocator, maybe we should just create a worker?

Reply:
maybe we can adjust this with how track works?

Reply:
the other hacky solution would be to block all deallocations until the raster task is finished.

Reply:
I guess we can't really do that. None of the other APIs require us to give up ownership of a texture to render it.

Reply:
We can avoid all of this by adding `ConcurrentTaskRunner::RunNowOrPostTask`.